doc: update SDC known issues (#10896)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

In addition to the known issues listed here, see also :ref:`softdevice_controller_limitations` for permanent limitations.

.. rst-class:: v2-3-0

DRGN-18651: Scanner may fail to deliver advertising reports

>   An edge case could be triggered if an advertising report is received just before scanner is timed out or disabled. This can lead to scanner not delivering any further advertising reports when re-enabled.

Reviewer: ```suggestion
  There is a small chance of failure if an advertising report is received just before the scanner times out or is disabled.
  This can lead to the scanner not delivering any further advertising reports when re-enabled.
```
